libcxx/test/std/utilities/memory/pointer.conversion/pointer_traits.pass.cpp

```diff
@@ -0,0 +1,176 @@
+//===----------------------------------------------------------------------===//
```

Comment:
These tests look really nice, but we should merge them with the ones in `libcxx/test/std/utilities/memory/pointer.traits/*` (for example `libcxx/test/std/utilities/memory/pointer.traits/difference_type.pass.cpp`).

I think it might make sense to consolidate them in a single `libcxx/test/std/utilities/memory/pointer.traits/types.pass.cpp`.